tools/distpartitioning/dist_lookup.py

```diff
@@ -0,0 +1,289 @@
+import numpy as np
+import os
+import pyarrow
+import torch
+
+from pyarrow import csv
+from gloo_wrapper import alltoallv_cpu
+
+class DistLookupService:
+
+    def __init__(self, input_dir, ntype_names, id_map, rank, world_size):
+        '''
+        Service constructor used to create an instance of this class.
+
+        Parameters:
+        -----------
+        self : instance of this class
+            instance of this class which is implicitly passed in every function call
+        input_dir : string
+            string representing the input directory where the node-type partition-id 
+            files are located
+        ntype_names : list of strings
+            list of strings which are used to read files located within the input_dir
+            directory and these files contents are partition-id's for the node-ids which
+            are of a particular node type
+        id_map : dgl.distributed.id_map instance
+            this id_map is used to retrieve ntype-ids, node type ids, and type_nids, per type
+            node ids, for any given global node id
+        rank : integer
+            integer indicating the rank of a given process
+        world_size : integer
+            integer indicating the total no. of processes
+        '''
+
+        assert os.path.isdir(input_dir)
+        assert ntype_names is not None
+        assert len(ntype_names) > 0
+
+        #these lists are indexed by ntype_ids
+        type_nid_begin = []
+        type_nid_end = []
+        partid_list = []
+        ntype_count = []
+
+        #iterate over the node types and extract the partition id mappings
+        for ntype in ntype_names:
+            print('[Rank: ', rank, '] Reading file: ', os.path.join(input_dir, '{}.txt'.format(ntype)))
+            df = csv.read_csv(os.path.join(input_dir, '{}.txt'.format(ntype)), \
+                read_options=pyarrow.csv.ReadOptions(autogenerate_column_names=True), \
+                parse_options=pyarrow.csv.ParseOptions(delimiter=' '))
+            ntype_partids = df['f0'].to_numpy()
+            count = np.int64(len(ntype_partids))
+            ntype_count.append(count)
+
+            #each rank assumes a contiguous set of partition-ids which are equally split
+            #across all the processes.
+            split_size = np.int64(np.ceil(count/np.int64(world_size)))
+            start, end = np.int64(rank * split_size), np.int64((rank+1)*split_size)
```

Comment:
are these type casts necessary?

Reply:
Removed some of the type_casts now... where it is NOT necessary.